### PR TITLE
feat(did): Add DID URL utilities

### PR DESCRIPTION
**Pull request by ender503**
- isDidUrl(): checks if given URL is a DID URL.
- decodeFromDidUrl(): decode information of the DID URL.

See also: https://w3c-ccg.github.io/did-spec/#fragment

### CODE REVIEW

File: packages/did/src/did.ts [hunk 1]
```diff
@@ -36,6 +39,39 @@ export const decodeFromDid = (tangleid: Did): MnidModel => {
...
 };
 
+/**
+ * Checks if given URL is a DID URL.
+ * @function isDidUrl
+ * @param {string} url - the URL to be checked.
+ * @return {boolean} The boolean that represent the url whether DID URL.
+ */
+export const isDidUrl = (url: string): boolean => {
+  return DID_URL_REGEX.test(url);
```
Comment (jserv) [inline packages/did/src/did.ts]: Is `test` a well-known method name? I would expect "validate".

Comment (jserv) [inline packages/did/src/did.ts]: Should we check from a known list of existing methods?

Comment (ender503) [reply packages/did/src/did.ts]: Check the method whether exist or not would be handled by another function, but we don't have the use case for that function for now.

File: packages/did/src/did.ts [hunk 2]
```diff
@@ -36,6 +39,39 @@ export const decodeFromDid = (tangleid: Did): MnidModel => {
...
+ * @return {{did: string, method: string, idstring: string, fragment: string}}
+ *   The information of the DID URL. {@link https://w3c-ccg.github.io/did-spec/#generic-did-syntax}
+ */
+export const decodeFromDidUrl = (url: string) => {
+  const matches = DID_URL_REGEX.exec(url);
+  if (matches === null) {
+    throw new Error('Invalid DID URL');
+  }
+
+  const [, method, idstring, fragment] = matches;
```
Comment (jserv) [inline packages/did/src/did.ts]: I wonder if there are restrictions over `idstring`, such as case sensitivity.

Comment (ender503) [reply packages/did/src/did.ts]: According to [Generic DID Syntax](https://w3c-ccg.github.io/did-spec/#generic-did-syntax), the restrictions are:
 - `idstring`: `A-Za-z0-9.-_`
 - `method`: `a-z0-9`

I rewrote the regular expression and make it more strictly.

**Pull request merged**